src/microbe_stage/PlayerHoverInfo.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using Godot;
+
+/// <summary>
+///   A system that manages reading what the player is hovering over.
+/// </summary>
+public class PlayerHoverInfo : Node
+{
+    private MicrobeCamera camera;
+    private CompoundCloudSystem cloudSystem;
+
+    /// <summary>
+    ///   All compounds the user is hovering over.
+    /// </summary>
+    public Dictionary<Compound, float> HoveredCompounds { get; private set; } = new Dictionary<Compound, float>();
+
+    /// <summary>
+    ///   All microbes the user is hovering over.
+    /// </summary>
+    public List<Microbe> HoveredMicrobes { get; private set; } = new List<Microbe>();
+
+    public bool IsHoveringAnyEntity => HoveredCompounds.Count > 0 || HoveredMicrobes.Count > 0;
```

Comment:
Are compound clouds actually considered entity? If not, this name might be misleading, but if it's clear that compound clouds can fall under this name I'm happy with it.

Reply:
Technically the individual clouds aren't but the game objects holding the fluid simulations technically are (at least I think I've called them entities in a comment or two). `IsHoveringOverAnything` might be a more neutral name (and suitable for https://github.com/Revolutionary-Games/Thrive/issues/1710 as well)